examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,579 @@
+model_2_scores = model_scores.iloc[1].values  # scores of the second-best model
+
+differences = model_1_scores - model_2_scores
+
+n = differences.shape[0]  # number of test sets
+df = n - 1
+n_train = len(list(cv.split(X, y))[0][0])
+n_test = len(list(cv.split(X, y))[0][1])
+
+t_stat, p_val = compute_corrected_ttest(differences, df, n_train, n_test)
+print(f"Corrected t-value: {t_stat:.3f}\n"
+      f"Corrected p-value: {p_val:.3f}")
+
+# %%
+# We can compare the corrected t- and p-values with the uncorrected ones:
+
+t_stat_uncorrected = (
+    np.mean(differences) / np.sqrt(np.var(differences, ddof=1) / n)
+)
+p_val_uncorrected = t.sf(np.abs(t_stat_uncorrected), df)
+
+print(f"Uncorrected t-value: {t_stat_uncorrected:.3f}\n"
+      f"Uncorrected p-value: {p_val_uncorrected:.3f}")
+
+# %%
+# Using the conventional significance alpha level at `p=0.05`, we observe that
+# the uncorrected t-test concludes that the first model is significantly better
+# than the second.
+#
+# With the corrected approach, in contrast, we fail to detect this difference.
+#
+# In the latter case, however, the frequentist approach does not let us
+# conclude that the first and second model have an equivalent performance. If
+# we wanted to make this assertion we need to use a Bayesian approach.
+
+# %%
+# Comparing two models: Bayesian approach
+# ---------------------------------------
+# We can use Bayesian estimation to calculate the probability that the first
+# model is better than the second. Bayesian estimation will output a
+# distribution that specifies the probability of each parameter value, which in
+# this example is the mean of the differences in the performance of two models.
+#
+# To obtain the posterior distribution we need to define a prior that models
+# our beliefs of how the means are distributed before looking at the data,
+# and multiply it by a likelihood function that computes how likely our
+# observed mean differences are, given the values that the mean of differences
+# could take.
+#
+# Bayesian estimation can be carried out in many forms to answer our question,
+# but in this example we will implement the approach suggested by Benavoli and
+# collegues [4]_.
+#
+# One way of defining our posterior using a closed-form expression is to select
+# a prior conjugate to the likelihood function. Benavoli and collegues [4]_
+# show that when comparing the performance of two classifiers we can model the
+# prior as a Normal-Gamma distribution (with both mean and variance unknown)
+# conjugate to a normal likelihood, to thus express the posterior as a normal
+# distribution.
+# Marginalizing out the variance from this normal posterior, we can define the
+# posterior of the mean parameter as a Student's t-distribution. Specifically:
+#
+# .. math::
+#    St(\mu;n-1,\overline{x},(\frac{1}{n}+\frac{n_{test}}{n_{train}})
+#    \hat{\sigma}^2)
+#
+# where :math:`\mu` represents the posterior of the mean difference in
+# performance,
```

Comment:
strictly speaking, I believe that the posterior is the distribution, not mu itself?

If we go with my suggested wording above where mu is defined, we probably don't need to specify it again here

Reply:
The posterior is the distribution of the mean of the differences.